buildSrc/src/main/kotlin/BuildInfo.kt

```diff
@@ -80,6 +127,194 @@ open class BuildInfo(project: Project) {
+  ): Iterable<TaskProvider<out Task>> =
+    with(project) {
+      val isMultiVendor = testJdkVendors.count() > 1
+      val basename = baseNameProvider()
+
+      serviceOf<JavaToolchainService>().let { toolchains ->
+        jdkTestRange
+          .asSequence()
+          .flatMap { targetVersion ->
+            // multiply out by jdk vendor
+            testJdkVendors.map { vendor -> (targetVersion to vendor) }
+          }
+          .filter { (version, _) ->
+            // unless we are instructed to test all JDKs, tests only include LTS releases and
+            // versions above the toolchain version.
+            testAllJdks || (JavaVersionRange.isLTS(version) || version >= jdkToolchainVersion)
+          }
+          .map { (jdkTarget, vendor) ->
+            if (jdkToolchainVersion == jdkTarget)
+              tasks.register(namer(jdkTarget, vendor)) {
+                // alias to `test`
+                dependsOn(basename)
+                group = Category.VERIFICATION
+                description = "Alias for regular '$basename' task, on JDK ${jdkTarget.asInt()}"
+              }
+            else
+              tasks.register(namer(jdkTarget, vendor.takeIf { isMultiVendor }), Test::class) {
+                group = Category.VERIFICATION
+                description = "Run tests against JDK ${jdkTarget.asInt()}"
+                configurator(toolchains.launcherFor { languageVersion = jdkTarget })
+
+                // fix: on jdk17, we must force the polyglot module on to the modulepath
+                if (jdkTarget.asInt() == 17)
+                  jvmArgumentProviders.add(
+                    CommandLineArgumentProvider {
+                      buildList { listOf("--add-modules=org.graalvm.polyglot") }
+                    }
+                  )
+              }
+          }
+          .filter {
+            // only include experimental tasks in the return suite if the flag is set. if the task
+            // is withheld from the returned list, it will not be executed by default with `gradle
+            // check`.
+            includeExperimental || !it.name.contains("Experimental")
+          }
+          .toList()
+      }
+    }
+
+  val javaCompiler: Provider<JavaCompiler> by lazy {
+    project.serviceOf<JavaToolchainService>().let { toolchainService ->
+      toolchainService.compilerFor { pklJdkToolchain() }
+    }
+  }
+
+  val javaTestLauncher: Provider<JavaLauncher> by lazy {
+    project.serviceOf<JavaToolchainService>().let { toolchainService ->
+      toolchainService.launcherFor { pklJdkToolchain() }
+    }
+  }
+
+  val multiJdkTesting: Boolean by lazy {
+    // By default, Pkl is tested against a full range of JDK versions, past and present, within the
+    // supported bounds of `PKL_TEST_JDK_TARGET` and `PKL_TEST_JDK_MAXIMUM`. To opt-out of this
+    // behavior, set `-DpklMultiJdkTesting=false` on the Gradle command line.
+    System.getProperty("pklMultiJdkTesting")?.toBoolean() ?: true
```

Comment:
To be kinder for the local dev experience, I think let's default this to false, then add `-DpklMultiJdkTesting=true` in CI

```suggestion
    System.getProperty("pklMultiJdkTesting")?.toBoolean() ?: false
```

Reply:
Whoops, this was supposed to be `false` all along. Good catch